src/ExcelProvider.Runtime/ExcelProvider.Runtime.fs

```diff
@@ -348,10 +368,22 @@ type Row(documentId, sheetname, rowIndex, getCellValue: int -> int -> obj, colum
         let value = this.GetValue columnIndex
 
         try
-            (value :?> Nullable<'a>).GetValueOrDefault()
-        with :? InvalidCastException ->
+            match value with
+            | null -> (value :?> Nullable<'a>).GetValueOrDefault()
+            | _ when typeof<'a> = typeof<string> -> (box (string value)) :?> 'a
+            //| :? string as valueStr when typeof<'a> = typeof<double> -> (box (Double.Parse valueStr)) :?> 'a
+            | :? double as valueDbl when typeof<'a> = typeof<DateTime> -> (box (DateTime.FromOADate valueDbl)) :?> 'a
+            | :? string as valueStr when typeof<'a> = typeof<DateTime> -> (box (DateTime.Parse valueStr)) :?> 'a
+            | _ -> value :?> 'a
+
+        with
+        | :? InvalidCastException
+        | :? ArgumentException
+        | :? ArgumentNullException
+        | :? FormatException ->
```

Comment:
These are copied here and probably (possibly?) a good candidate for splitting out as their own function.